tools/Jenkins/dependency_spotter.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import re
+import urllib.request
+
+
+def dependency_spotter(os_name: str, first_build: int, second_build: int, pipeline: str, log_file: str):
```

Comment:
Would be good to add doc strings on all functions (see `generate_cppcheck_suppressions_list.py` for examples).